### PR TITLE
Added an option to open desktop folders in default file manager

### PR DESCRIPTION
**Pull request by tsujan**
The option is in the first tab of Desktop Preferences and is unchecked by default. It has effect when desktop folders are launched in the usual way, i.e., by left clicking, and the text label is explicit about that. Middle/right clicking triggers the default behavior.

NOTE: For user convenience, the home shortcut will also be opened by the default file manager if the option is checked but the other shortcuts are always opened by pcmanfm-qt because other file managers may not be able to handle them.

Closes https://github.com/lxqt/lxqt/issues/1823

### CODE REVIEW

**Comment by luis-pereira** [comment]
Will test.

File: pcmanfm/launcher.cpp
Comment (luis-pereira) [inline pcmanfm/launcher.cpp:52]: openWithDefaultFileManager is always false.

**Review by luis-pereira — COMMENTED** [review]
It's not working for me.

- Setup the default file manager to something else
- Checked the use default file manager
- Left clicked on a desktop folder. It opens with pcmanfm-qt.

Traced it to openWithDefaultFileManager being always false (launcher.cpp:52). Run out of time to find the cause.

File: pcmanfm/launcher.cpp
Comment (tsujan) [reply pcmanfm/launcher.cpp:52]: Strange! It works here. But there is a possibility that I should investigate; will tell you soon. Thanks for testing and reviewing!

Comment (tsujan) [reply pcmanfm/launcher.cpp:52]: @luis-pereira 
Apart from my installation, that works, I downloaded the branch ["default_file_manager"](https://github.com/lxqt/pcmanfm-qt/tree/default_file_manager), compiled it and tested it without installation on two machines. In all cases, desktop folders were opened by Dolphin when I checked "Open desktop folders in…" and set the default file manager to Dolphin. All changes were applied on-the-fly and correctly.

I have no idea why your test resulted in a different behavior.

Comment (tsujan) [reply pcmanfm/launcher.cpp:52]: At the risk of stating the obvious, OK or Apply should be pressed for changes to take effect.

**Comment by stefonarch** [comment]
It works here, with thunar set to default file manager in "file associations, default applications" tab.

> Middle/right clicking triggers the default behavior.

Was it supposed to open with pcmanfm-qt? Middle click on folders does nothing here.

**Comment by luis-pereira** [comment]
@tsujan I'm I doing something wrong ? https://youtu.be/lRDpWY0btVA

**Comment by tsujan** [comment]
> @tsujan I'm I doing something wrong ? https://youtu.be/lRDpWY0btVA

They say a video on YouTube is worth 10 thousand words ;) In the video, You click on a desktop entry (with `Exec=pcmanfm-qt computer:///`), not on folder. 

I didn't cover desktop entries because we can't suppose all file managers can open them (for example, a file manager may not recognize `computer:///`). The patch is only for folders.

> Was it supposed to open with pcmanfm-qt? 

@stefonarch, yes, it does open pcmanfm-qt here.

**Comment by luis-pereira** [comment]
@tsujan Ok. Check this one https://youtu.be/T81GpIqXHbc

**Comment by tsujan** [comment]
@luis-pereira 

Again, the second item you clicked in your second video is a desktop entry with `Exec=pcmanfm-qt /home/YOUR_USER`.

There is a risk I don't want to take: If I changed the exec commands of desktop entries and removed `pcmanfm-qt` from them, this new feature could damage the default behavior (because those entries could be at the mercy of another file manager, *even without the option*).

On the other hand, making another set of desktop entries just for this option would be too much.

In my opinion, this patch is what can be done *safely* and *frugally* for a feature that I consider unimportant. I'll be happy if the parch is NOT merged (but won't be sad if it is). However, I can't agree with more that this.

**Comment by tsujan** [comment]
I suggest we don't merge it and, instead, wait to see if users of LXQt ≥ 0.16.0 will need the feature in question. If it's proved to be popular, it'll be worth more changes, which could be made to the current PR.

**Comment by luis-pereira** [comment]
@tsujan Thanks for your thorough explanation.
Will think about it and come back on this.

**Comment by luis-pereira** [comment]
@tsujan I would be "happy" if it's merged.
For the Trash, Computer and Network use pcmanfm is right, since not all file managers support it.
But for the Home folder we could use the default file manager.  What's your opinion ?
A very rough implementation at https://github.com/lxqt/pcmanfm-qt/pull/1219.

**Comment by tsujan** [comment]
@luis-pereira 
I'm happy that you work on it too (https://github.com/lxqt/pcmanfm-qt/pull/1219). IMHO, this needs more code. We need to change all shortcuts whenever the option is checked/unchecked. Also, pcmanfm-qt's desktop can be used outside LXQt (I even tested it under Wayfire — it worked fine; the problems were Wayfire's). We should also do something about `computer:///` because, for example, Dolphin doesn't understand it; etc.

Let's do it after 0.16.0; we don't have time for tests now.

**Comment by tsujan** [comment]
Oh, sorry, I forgot this:

> But for the Home folder we could use the default file manager. What's your opinion ?

Yes, that's a good plan and [your PR](https://github.com/lxqt/pcmanfm-qt/pull/1219) seems to do it right; `qtxdg-mat` is the best opion. We just need to overwrite the home shortcut if the option is changed while it exists.

**Comment by tsujan** [comment]
@luis-pereira 
I added a commit and squashed. It passed my tests with Dolphin, Nautilus, Thunar and spacefm. Please test it!

**Comment by tsujan** [comment]
BTW, in https://github.com/lxqt/pcmanfm-qt/pull/1219#issuecomment-718246131, I said, "... because a file manager should be able to open links". I was drowsy then ;) The link is handled by `libfm-qt` and its *folder* is opened by the default file manager. Therefore, the default file manager doesn't even need to know what a link is.

**Review by luis-pereira — APPROVED** [review]
Tested with spacefm, thunar, dolphin and.... get this... pcmanfm-qt ;)
Works as described.

**Comment by tsujan** [comment]
@luis-pereira 
Thanks! Now that we agree on the home shortcut, what do you think about merging? Should we exclude it from 0.16.0?

**Comment by stefonarch** [comment]
Tested too, works fine. I wonder if "Computer" folder  should open too in the default filemanager, to have consistent behaviour with "Home" folder.

**Comment by tsujan** [comment]
> Tested too

Thanks!

> I wonder if "Computer" folder should open too

Dolphin doesn't understand `computer:///` and we can't assume other file managers do. The only shortcut that all file managers can handle is home because it gives an ordinary folder to them.

**Comment by stefonarch** [comment]
Clear, didn't know this. Updating translations after merge please - weblate is sending mail notifications for new string to translate :)

**Comment by tsujan** [comment]
> Updating translations after merge please

I'll make PRs for all LXQt components to update changelogs and translations, starting from tomorrow. Did the same thing for 0.15.0 and will do it before all major releases.

As for merging this PR, let's wait for @luis-pereira's reply.

**Comment by luis-pereira** [comment]
> As for merging this PR, let's wait for @luis-pereira's reply.

@tsujan I approved after testing it (12 hours ago). 
We can continue to work on it.But the current status is an good improvement  and works fine. Ship it.

**Comment by tsujan** [comment]
> I approved after testing it (12 hours ago).

Yes, I knew that, of course. My question was about the lack of time for more tests or, in other words, "feature freeze" (https://github.com/lxqt/lxqt/issues/1848#issuecomment-714323140).

If something goes wrong, a point release is possible.

**Comment by luis-pereira** [comment]
> Yes, I knew that, of course. My question was about the lack of time for more tests or, in other words, "feature freeze" ([lxqt/lxqt#1848 (comment)](https://github.com/lxqt/lxqt/issues/1848#issuecomment-714323140)).

You tested, I tested. No obvious issue.
If something is wrong we will get reports. 
 
> If something goes wrong, a point release is possible.

Of course.

**Comment by tsujan** [comment]
> You tested, I tested

True. @stefonarch tested it too. My cautiousness may seem irrational but I've seen too many hidden bugs.

Anyway, we agreed on point releases in April :)

**Pull request merged**
